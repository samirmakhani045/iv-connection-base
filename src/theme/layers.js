export const modal = 500;
export const popover = 200;
export const cookieBanner = 100;
export const aboveBackground = 1;
export const background = -1;
