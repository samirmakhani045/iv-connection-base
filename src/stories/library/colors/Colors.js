import React from 'react';

import * as colors from '@studio/theme/colors';

import { Wrapper, ColorWrapper, ColorSquare, ColorName } from './Colors.styles';

const Colors = () => {
  return (
    <>
      <h1>Colors</h1>
      <Wrapper>
        {Object.entries(colors).map(([colorName, colorValue]) => (
          <ColorWrapper key={colorName}>
            <ColorSquare color={colorValue} hasBorder={colorName.startsWith('white')} />
            <ColorName>{colorValue}</ColorName>
            <ColorName>{colorName}</ColorName>
          </ColorWrapper>
        ))}
      </Wrapper>
    </>
  );
};

export default Colors;
