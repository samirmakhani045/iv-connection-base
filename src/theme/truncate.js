import * as media from '@studio/theme/breakpoints';

export const multiline = (xs = 1, s, m, l) => {
  const maxLinesS = s || xs;
  const maxLinesM = m || maxLinesS;
  const maxLinesL = l || maxLinesM;

  return `
    overflow: hidden;
    display: -webkit-box; /* stylelint-disable-line value-no-vendor-prefix */
    -moz-box-orient: vertical; /* stylelint-disable-line value-no-vendor-prefix */
    -webkit-box-orient: vertical; /* stylelint-disable-line value-no-vendor-prefix */
    box-orient: vertical;
    -webkit-line-clamp: ${xs}; /* stylelint-disable-line value-no-vendor-prefix */

    ${media.smallMedia} {
      -webkit-line-clamp: ${maxLinesS};
    }

    ${media.mediumMedia} {
      -webkit-line-clamp: ${maxLinesM};
    }

    ${media.largeMedia} {
      -webkit-line-clamp: ${maxLinesL};
    }
  `;
};
