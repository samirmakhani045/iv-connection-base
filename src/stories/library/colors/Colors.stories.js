import React from 'react';

import { withPadding } from '@utils/decorators';

import Colors from './Colors';

export default {
  title: 'Library/Colors',
  component: Colors,
  decorators: [(story) => withPadding(story)],
};

export const Default = () => <Colors />;
