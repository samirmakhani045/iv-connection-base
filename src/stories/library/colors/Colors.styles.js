import styled from 'styled-components/macro';

import * as colors from '@studio/theme/colors';

export const Wrapper = styled.ul({
  display: 'flex',
  flexdirection: 'column',
  flexWrap: 'wrap',
  listStyle: 'none',
  margin: 0,
  padding: 0,
});

export const ColorWrapper = styled.li({
  alignItems: 'center',
  display: 'grid',
  gridTemplateColumns: '3rem 1fr 1fr',
  maxWidth: '40rem',
  width: '100%',
});

export const ColorSquare = styled.div((props) => ({
  backgroundColor: props.color,
  ...(props.hasBorder && { border: `0.1rem solid ${colors.cloud4}` }),
  height: '3rem',
  width: '3rem',
}));

export const ColorName = styled.span({
  color: colors.textSecondary,
  fontFamily: 'monospace',
  fontSize: '1rem',
  paddingLeft: '1rem',
});
