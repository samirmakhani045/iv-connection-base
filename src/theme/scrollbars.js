import * as colors from '@studio/theme/colors';

export default `
  /* styles firefox */
  * {
    scrollbar-width: thin;
    scrollbar-color: ${colors.textSecondaryA15} transparent;
  }

  /* styles chrome, edge and safari */
  *::-webkit-scrollbar {
    height: 1rem;
    width: 1rem;
  }

  *::-webkit-scrollbar-track {
    background: transparent !important;
  }

  *::-webkit-scrollbar-thumb {
    background-color: ${colors.textSecondaryA15};
    background-clip: content-box;
    border-radius: 1rem;
    border: 0.3rem solid transparent;
  }
`;
