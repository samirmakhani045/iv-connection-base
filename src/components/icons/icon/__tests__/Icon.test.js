import React from 'react';
import { render, screen } from '@testing-library/react';

import * as colors from '@studio/theme/colors';

import Icon from '../Icon';

const defaultProps = {
  name: 'Add',
};

const Test = (props) => <Icon {...defaultProps} {...props} />;

describe('<Icon />', () => {
  it('should render correctly', () => {
    const { container } = render(<Test />);

    expect(screen.queryByTestId('wrapper')).toBeInTheDocument();
    expect(screen.queryByTestId('icon')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should not render if icon does not exist', () => {
    render(<Test name="ThisIconDoesNotExist" />);

    expect(screen.queryByTestId('wrapper')).not.toBeInTheDocument();
  });

  it('should render a different icon, bigger and red', () => {
    const { container } = render(<Test name="Arrow" size={60} color={colors.red} />);

    expect(screen.queryByTestId('wrapper')).toBeInTheDocument();
    expect(screen.queryByTestId('icon')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
