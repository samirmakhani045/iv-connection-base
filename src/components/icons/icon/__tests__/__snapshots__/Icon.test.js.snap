// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should render a different icon, bigger and red 1`] = `
<div
  class="Iconstyles__IconWrapper-sc-1ghvgbo-0 kDPqTt"
  data-testid="wrapper"
  size="60"
>
  <svg
    data-testid="icon"
    fill="#E93A5A"
    height="1em"
    transform="rotate(0)"
    viewBox="0 0 24 24"
    width="1em"
  >
    <path
      d="M9.29492 16.885L13.8749 12.295L9.29492 7.70498L10.7049 6.29498L16.7049 12.295L10.7049 18.295L9.29492 16.885Z"
    />
  </svg>
</div>
`;

exports[`<Icon /> should render correctly 1`] = `
<div
  class="Iconstyles__IconWrapper-sc-1ghvgbo-0 jzZALt"
  data-testid="wrapper"
  size="24"
>
  <svg
    data-testid="icon"
    fill="#313336"
    height="1em"
    viewBox="0 0 24 24"
    width="1em"
  >
    <path
      d="M19 13H13V19H11V13H5V11H11V5H13V11H19V13Z"
    />
  </svg>
</div>
`;
